src/daft-minhash/src/minhash.rs

```diff
@@ -0,0 +1,215 @@
+use std::{
+    ops::{Add, BitAnd, Mul, Rem, Shr},
+    simd::{cmp::SimdOrd, Simd},
+};
+
+use common_error::DaftResult;
+use mur3::murmurhash3_x86_32;
+
+// which SIMD to use
+const SIMD_LANES: usize = 8;
+type S = Simd<u64, SIMD_LANES>;
```

Comment:
What's the best way to deal with the lane count? On my machine and the AWS EC2 instance I tested it on, the best value happened to be 8, but it seems reasonably likely that the optimal value changes depending on hardware. Should we provide some option to change this? (Also, 8 lanes seems strangely low, I was expecting 32 or 64 to be more optimal, but it was slower in testing...)